ui/app/src/components/LegacyDashboardFrame/LegacyDashboardFrame.tsx

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2007-2020 Crafter Software Corporation. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as published by
+ * the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { IframeHTMLAttributes } from 'react';
+import { createStyles, makeStyles } from '@material-ui/core/styles';
+import { useEnv } from '../../utils/hooks';
+
+interface LegacyDashboardFrameProps {
+  dashboardOnly?: boolean;
+  iframeProps?: IframeHTMLAttributes<any>;
+}
+
+const useStyles = makeStyles(() =>
+  createStyles({
+    iframe: {
+      width: '100%',
+      height: '100%',
+      border: 'none'
+    }
+  })
+);
+
+function LegacyDashboardFrame(props: LegacyDashboardFrameProps) {
+  const { dashboardOnly = true, iframeProps } = props;
+  const authoringUrl = useEnv().authoringBase;
+  const classes = useStyles();
+
+  const iframeSrc = `${authoringUrl}/site-dashboard${[dashboardOnly && '?mode=embedded'].filter(Boolean).join('')}`;
+
+  return <iframe frameBorder={0} title="Site Dashboard" src={iframeSrc} className={classes.iframe} {...iframeProps} />;
+}
```

Comment:
Too much code duplication between these two components.
Create LegacyIFrame and use it from both.